### PR TITLE
Add Cabal file spellchecker

### PR DESCRIPTION
**Pull request by zjhmale**
screenshot

### CODE REVIEW

**Comment by rikvdkleij** [comment]
Thanks!

File: src/main/scala/intellij/haskell/spellchecker/CabalSpellcheckingStrategy.scala [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package intellij.haskell.spellchecker
+
+import com.intellij.psi.PsiElement
+import com.intellij.spellchecker.tokenizer.{SpellcheckingStrategy, Tokenizer}
+import intellij.haskell.cabal.CabalLanguage
+
+/**
+  * Provide spellchecker support for Cabal sources.
+  */
+class CabalSpellcheckingStrategy extends SpellcheckingStrategy {
```
Comment (rikvdkleij) [inline src/main/scala/intellij/haskell/spellchecker/CabalSpellcheckingStrategy.scala]: Can you explain me how this works? I do not see a dictionary file.

Comment (zjhmale) [inline src/main/scala/intellij/haskell/spellchecker/CabalSpellcheckingStrategy.scala]: The document is [here](https://www.jetbrains.com/help/idea/2016.2/spellchecking.html), there is a built-in dictionary inside intellij, and [here](https://github.com/rikvdkleij/intellij-haskell/blob/master/src/main/resources/dictionary/haskell.dic) is a explcit dictionary for haskell words which we want to be accepted by the typo checker.

**Comment by rikvdkleij** [comment]
That dict file for Haskell words is also used by the Cabal spellchecker?

**Comment by zjhmale** [comment]
@rikvdkleij Yes, exactly, e.g. `aeson` will be a valid word in cabal file.
>That dict file for Haskell words is also used by the Cabal spellchecker?

**Pull request merged**